### PR TITLE
[READY] Makes TRAIT_STUNRESISTANCE actually work

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Right now TRAIT_STUNRESISTANCE only makes batons knock you down for 0.5 seconds instead of 5, while you still get as much damage, allowing security to still stamcrit you with 2 hits. In this PR, amount of stamina damage taken by people with TRAIT_STUNRESISTANCE was reduced by 33%, making it actually useful. Flashes, peppersprays, shoves and other sources of stuns and stamina damage were unchanged and you still can stun people with adrenals/stimpacks that way.

## Why It's Good For The Game

Currently there's no real counter to stunbatons other than hulks. There's also zero sense in buying traitor stimpack/ling adrenals because their stun resist didn't work

## Changelog
:cl:
balance: Stun resistance applied by certain chems and abilities now lowers stam damage from batons
/:cl:

### CODE REVIEW

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -233,10 +233,19 @@
 			return FALSE
 	/// After a target is hit, we do a chunk of stamina damage, along with other effects.
 	/// After a period of time, we then check to see what stun duration we give.
-	L.Jitter(20)
-	L.set_confusion(max(confusion_amt, L.get_confusion()))
-	L.stuttering = max(8, L.stuttering)
-	L.apply_damage(stamina_loss_amt, STAMINA, BODY_ZONE_CHEST)
+	var/trait_check = HAS_TRAIT(target, TRAIT_STUNRESISTANCE)
```
Comment (Iamgoofball) [inline code/game/objects/items/stunbaton.dm:236]: how did you manage to clone this locally, make all these changes, and then forget to try to compile it?

**Review by Iamgoofball — CHANGES_REQUESTED** [review]
Your code doesn't compile.

**Comment by kriskog** [comment]
This trait does what it was intended to, as its available from the pump-up injectors in maintenance, as well as the extremely simple recipe of epi and coffee. Its existence in other stimulant drugs was added there to also give them the knockdown time reduction. Note that these drugs also already give hardstun reductions in the chemical itself. There is no good reason to change what this trait does, especially when it is so easy to get. 

Personally I am of the opinion that stimpacks could probably use some tuning in the other direction than where you are taking them. I was fairly happy with where I left ling adrenals last I touched them.

**Comment by SmArtKar** [comment]
> This trait does what it was intended to, as its available from the pump-up injectors in maintenance, as well as the extremely simple recipe of epi and coffee. Its existence in other stimulant drugs was added there to also give them the knockdown time reduction. Note that these drugs also already give hardstun reductions in the chemical itself. There is no good reason to change what this trait does, especially when it is so easy to get.
> 
> Personally I am of the opinion that stimpacks could probably use some tuning in the other direction than where you are taking them. I was fairly happy with where I left ling adrenals last I touched them.

Adrenals and stimpack feel kinda useless because they just make you a bit faster and not actually reduce any stuns as they're supposed to do. If people think that they're fine as they currently are, I'll close this PR

**Comment by kriskog** [comment]
Yup, the stimpack does nothing to reduce stuns
image

**Comment by SmArtKar** [comment]
> Yup, the stimpack does nothing to reduce stuns
> image

I checked on local and you still get stamcritted from 2 baton hits and lie pretty much as long as you would without stimpack

**Comment by Sparkezel** [comment]
I think it would be nice for those things to be more useful gainst stuns, and I'm really tired of hulks

**Comment by kriskog** [comment]
Non-lethals need to be reliable enough that they see use over their lethal counterparts. The stimpack is already a "must buy" for many players, and they can be really hard to take down. I am not interested in making them even better.

Note that there are chems that increase your stamina regen which can drastically reduce the time you spend stunned, but these are not and should not be available from the uplink at a button press. Put some effort into getting anti stun stuff outside of the uplink if you need the stronger stuff to make up for your unrobustness, or work on getting robust.

**Comment by ATH1909** [comment]
> Note that these drugs also already give hardstun reductions in the chemical itself.

Actually, giving this trait is literally the only positive thing that pump-up does (unless you count satisfying maintenance drug addictions?).

File: code/game/objects/items/stunbaton.dm [hunk 2]
```diff
@@ -233,10 +233,19 @@
...
-	L.set_confusion(max(confusion_amt, L.get_confusion()))
-	L.stuttering = max(8, L.stuttering)
-	L.apply_damage(stamina_loss_amt, STAMINA, BODY_ZONE_CHEST)
+	var/trait_check = HAS_TRAIT(L, TRAIT_STUNRESISTANCE)
+
+	if(trait_check)
+		L.Jitter(5)
+		L.set_confusion(max(confusion_amt * 0.2, L.get_confusion()))
+		L.stuttering = max(2, L.stuttering)
+		L.apply_damage(stamina_loss_amt * 0.2, STAMINA, BODY_ZONE_CHEST)
```
Comment (ATH1909) [inline code/game/objects/items/stunbaton.dm:242]:
```suggestion
		L.apply_damage(stamina_loss_amt * 2/3, STAMINA, BODY_ZONE_CHEST)
```
perhaps 40 stamina damage could serve as a compromise value.

**Comment by ATH1909** [comment]
Since you're touching TRAIT_STUNRESISTANCE anyway, could you rename it to TRAIT_BATONRESISTANCE (or maybe TRAIT_BATON_RESISTANCE?) so that its name is less misleading?

Making it work against classic batons as well would also help quite a bit.

**Comment by kriskog** [comment]
> I think it would be nice for those things to be more useful gainst stuns, and I'm really tired of hulks

Worth noting that I am a heavy advocate for just removing hulk, and have been for years. People becoming hulks are a symptom of hulk not getting deleted yet, and should not be brought in as an argument in every discussion about everything, thanks. :)

> Making it work against classic batons as well would also help quite a bit.

Classic batons are already kinda hot trash. 

> If you really wanted to push pump-up, ephedrine, and the like, you could also make TRAIT_STUNRESISTANCE completely remove the knockdown from (stun) baton strikes, but it looks like skog is on the fence as it is. Skog, would you be okay with this knockdown removal if TRAIT_STUNRESISTANCE was removed from the chems from stimpacks and changeling adrenals?

No. Look to the baton changes that were proposed recently, and see what they were about and what happened to them.

**Comment by Sparkezel** [comment]
>Worth noting that I am a heavy advocate for just removing hulk, and have been for years. People becoming hulks are a symptom of hulk not getting deleted yet, and should not be brought in as an argument in every discussion about everything, thanks. :)

Hulk should be a traitor item AT BEST, fermichoom made good ol' chooms much less accessible by now and baton being the pretty much BEST weapon in game is very sad :)
Alas I have no idea how to do any of that better than it is either... still though, in my opinion anti stun medication could use a little buff 😄

**Comment by maxymax13** [comment]
trait_susresistance

**Comment by MoonFalcon** [comment]
> This trait does what it was intended to, as its available from the pump-up injectors in maintenance, as well as the extremely simple recipe of epi and coffee. Its existence in other stimulant drugs was added there to also give them the knockdown time reduction. Note that these drugs also already give hardstun reductions in the chemical itself. There is no good reason to change what this trait does, especially when it is so easy to get.
> 
> Personally I am of the opinion that stimpacks could probably use some tuning in the other direction than where you are taking them. I was fairly happy with where I left ling adrenals last I touched them.

Pump ups currently do almost nothing for stun batons, and will not help you at all in combat. Pump ups are COMPLETELY USELESS right now and are 100% bloat. There is 100% a reason to change what these do as there is currently no in game effect. Try it out yourself!

Stimpacks used to be used as a stun baton counter. We need to go back to that. After the stamina changes to stun batons stimpacks became a way just to move fast. 


> Non-lethals need to be reliable enough that they see use over their lethal counterparts. The stimpack is already a "must buy" for many players, and they can be really hard to take down. I am not interested in making them even better.
> 
> Note that there are chems that increase your stamina regen which can drastically reduce the time you spend in stam crit, but these are not and should not be available from the uplink at a button press. Put some effort into getting anti stun stuff outside of the uplink if you need the stronger stuff to make up for your unrobustness, or work on getting robust.

who the fuck buys the stimpack lmao, it's only purpose is to make you to temporarily move fast, a complete waste of TC when you can make 10 other chems to do that. STIMPACK DOES NOT MEANINGFULLY REDUCE THE TIME YOU SPEND IN STAM CRIT FROM A STUN BATON. TRY IT YOURSELF! There is currently no antagonist counter to stun batons other than some wizard spells. Traitors and lings have 0 counter to them.

The current state is that a single tool can defeat any antagonist and is easily made by anyone on the station. The fact that they take two hits means the crew has an advantage vs the antagonist. It's completely imbalanced.

**Comment by MoonFalcon** [comment]
> > I think it would be nice for those things to be more useful gainst stuns, and I'm really tired of hulks
> 
> Worth noting that I am a heavy advocate for just removing hulk, and have been for years. People becoming hulks are a symptom of hulk not getting deleted yet, and should not be brought in as an argument in every discussion about everything, thanks. :)
> 
> > Making it work against classic batons as well would also help quite a bit.
> 
> Classic batons are already kinda hot trash.
> 
> > If you really wanted to push pump-up, ephedrine, and the like, you could also make TRAIT_STUNRESISTANCE completely remove the knockdown from (stun) baton strikes, but it looks like skog is on the fence as it is. Skog, would you be okay with this knockdown removal if TRAIT_STUNRESISTANCE was removed from the chems from stimpacks and changeling adrenals?
> 
> No. Look to the baton changes that were proposed recently, and see what they were about and what happened to them.

Classic batons are BETTER than normal batons. They both require two hits and then put you in stam crit, except classic batons knock you down instantly rather than after a few seconds

**Comment by Fhaxaris** [comment]
> The stimpack is already a "must buy" for many players, and they can be really hard to take down.

I've never seen it be purchased post the baton nerf, where is this coming from.
Have you ever used it? It's near worthless, the time it takes for you to get out of stam crit from two strikes is identical, your stamina will stay above 100 before it resets, meaning it did nothing at all.
It only soft counters slips and true stuns, not the stamina stick or disabler.

**Comment by ATH1909** [comment]
> > > I think it would be nice for those things to be more useful gainst stuns, and I'm really tired of hulks
> > 
> > 
> > Worth noting that I am a heavy advocate for just removing hulk, and have been for years. People becoming hulks are a symptom of hulk not getting deleted yet, and should not be brought in as an argument in every discussion about everything, thanks. :)
> > > Making it work against classic batons as well would also help quite a bit.
> > 
> > 
> > Classic batons are already kinda hot trash.
> > > If you really wanted to push pump-up, ephedrine, and the like, you could also make TRAIT_STUNRESISTANCE completely remove the knockdown from (stun) baton strikes, but it looks like skog is on the fence as it is. Skog, would you be okay with this knockdown removal if TRAIT_STUNRESISTANCE was removed from the chems from stimpacks and changeling adrenals?
> > 
> > 
> > No. Look to the baton changes that were proposed recently, and see what they were about and what happened to them.
> 
> Classic batons are BETTER than normal batons. They both require two hits and then put you in stam crit, except classic batons knock you down instantly rather than after a few seconds

classic batons have a 4 second cooldown compared to a baton's 2.5 second cooldown, though. however, you can get around that by just using two classic batons (or one classic baton and one stun baton) because classic batons keep track of their cooldowns independently lmaoooooo

File: code/game/objects/items/stunbaton.dm [hunk 3]
```diff
@@ -233,10 +233,19 @@
 			return FALSE
 	/// After a target is hit, we do a chunk of stamina damage, along with other effects.
 	/// After a period of time, we then check to see what stun duration we give.
-	L.Jitter(20)
-	L.set_confusion(max(confusion_amt, L.get_confusion()))
-	L.stuttering = max(8, L.stuttering)
-	L.apply_damage(stamina_loss_amt, STAMINA, BODY_ZONE_CHEST)
+	var/trait_check = HAS_TRAIT(L, TRAIT_STUNRESISTANCE)
+
+	if(trait_check)
```
Comment (Mothblocks) [inline code/game/objects/items/stunbaton.dm:238]: You don't need the variable, just run HAS_TRAIT in here.

**Comment by kriskog** [comment]
> Pump ups currently do almost nothing for stun batons, and will not help you at all in combat. Pump ups are COMPLETELY USELESS right now and are 100% bloat. There is 100% a reason to change what these do as there is currently no in game effect. Try it out yourself!

It reduces your knockdown from stun batons, sufficient effect from a free garbage injector in maint.

As for the rest of your comment, I guess get good. If you need 80% damage reduction to achieve something as traitor, this might not be the game for you.

**Comment by MoonFalcon** [comment]
> > Pump ups currently do almost nothing for stun batons, and will not help you at all in combat. Pump ups are COMPLETELY USELESS right now and are 100% bloat. There is 100% a reason to change what these do as there is currently no in game effect. Try it out yourself!
> 
> It reduces your knockdown from stun batons, sufficient effect from a free garbage injector in maint.
> 
> As for the rest of your comment, I guess get good. If you need 80% damage reduction to achieve something as traitor, this might not be the game for you.

Not sure what you are on about with the 80% thing, all I want is for there not to be a single, easily manufactured tool that can instantly defeat any traitor/ling/heretic. Especially when those mentioned antagonists have no real counter for it (other than hulk).

The current tg meta is: baton. That's it, that's the entire meta. And it's awful.

**Comment by Fhaxaris** [comment]
> I guess get good. If you need 80% damage reduction to achieve something as traitor, this might not be the game for you.

Good refutation. Anyone who wants an item to have more utility than a floor tile is a bad player and needs to get good.

**Comment by kriskog** [comment]
> The knockdown difference is what, half a second? And how does that matter when they will just immediately follow up? My point is that they are useless. No one is using them. They are bloat. Either remove them or fix them. Same with adrenals and stimpacks.

Non chem assisted the baton knockdown is 5 seconds, vs 0.5 with the trait. There is a significant difference, if you arent using it you should be.
Im more than happy to remove both ling adrenals and stimpacks.

> Not sure what you are on about with the 80% thing, all I want is for there not to be a single, easily manufactured tool that can instantly defeat any traitor/ling/heretic. Especially when those mentioned antagonists have no real counter for it (other than hulk).

This PR wanted to make the stun resist trait into an 80% stamina damage reduction. 

> antagonists have no real counter for it

You do. Dont get hit. There are a lot of ranged options available to deal with stun batons, you dont have to be in securitys face with a welding tool or whatever. \

Just in case you are too new to know, there used to be a hard counter to stuns in the uplink through the old adrenal implant. Hard counters to stuns make the shotguns come out and be a necessary part of the security kit, it's bad for the game overall.

**Comment by Mothblocks** [comment]
> This trait does what it was intended to

@kriskog Are you confident on this? If so, why have you not closed it?

**Comment by kriskog** [comment]
24 hours

**Comment by Mothblocks** [comment]
well at least dnm label it then

**Comment by MoonFalcon** [comment]
> Non chem assisted the baton knockdown is 5 seconds, vs 0.5 with the trait. There is a significant difference, if you arent using it you should be.

Just tested it and confirmed this, but how does this help when the recharge time for a baton is shorter than the time it takes for you to get back up. Its the exact same in terms of gameplay.

> Im more than happy to remove both ling adrenals and stimpacks.

I don't doubt it in the least

>You do. Dont get hit. There are a lot of ranged options available to deal with stun batons, you dont have to be in securitys face with a welding tool or whatever.

So the solution to the infalliable stun baton, killer of worlds, defeater of any antag, is to...
not get hit...
So if thats the solution, why did we remove old tasers? why nerf old batons? why remove shotguns? You can dodge any of those. Why did we replace a system where at least the antags could use the equipment to their advantage to a completely one sided system? What do you do when sec decides to random search you? What about when the entire crew is armed with batons? Are you just going to stay out of range of every single person forever? Should we remove melee combat from the game beside batons?

> Just in case you are too new to know, there used to be a hard counter to stuns in the uplink through the old adrenal implant. 
Hard counters to stuns make the shotguns come out and be a necessary part of the security kit, it's bad for the game overall.

I can't tell if I'm being baited into giving up who I am or what but the old adrenals for ling and traitor were amazing. You could actually have high octane melee combat moments because fights didn't end in ~1 second. I see nothing wrong with making sec actually use wide variety of other tools that never get touched. 

You clearly have made up your mind, which is a shame, because I really think this can put TG on track to a much better place combat wise. Also, you completely avoided my combat challenge. If you are so robust (you must be if you can tell others to get good) I don't see why this is an issue. If you can defeat me even once with any traitor weapon or ling ability, while all I have is a baton, I'll completely give up on commenting on this PR. If I win you have to merge it.

**Comment by MoonFalcon** [comment]
Combat should be damaged based, not stun based. Shotguns are great for this, spears and fireaxes and mosins and lasers are wonderful for this. They make for drawn out high octane fights that require a lot of skill.

Batons are the complete opposite of this. They end fights in one move, usually over the course of a couple seconds. Merging this can bring us to a better damaged based model as people won't be able to use batons as a one-size-fits-all solution for every single problem

**Comment by kriskog** [comment]
@MoonFalcon In our current state of the game, stuns have to exist or there is no non lethal play. We are not interested in the game turning into a lethal pvp deathmatch every time security has to make an arrest. Some times you make mistakes, the other side gets the advantage, and you lose. And that has to be ok, on both sides. If you want some 100% balanced pvp deathmach, you are playing the wrong game.

@SmArtKar I get where you are trying to go with this, but changing this trait is a clumsy approach. If you wish to be more resilient against stun batons and other stamina damage there are already chemicals that are available through chemistry that will do this, and that stack with the stimpack. Look at aranesp, for instance. Keep in mind that healing stamina damage between baton hits allow you to take more than the normal 2, which combined with the shorter knockdown gives you a pretty heavy advantage against a baton.

**Comment by Hamurlik** [comment]
remove stunbatons instead give security banana launchers to slip criminals

**Pull request closed without merging**